File: audio-analysis-worker/prompt_renderer.py
"""오디오 분석 서비스가 소유한 일반 텍스트 템플릿만 렌더링한다."""
from pathlib import Path
from jinja2 import Environment, FileSystemLoader, StrictUndefined

_ENVIRONMENT = Environment(
    loader=FileSystemLoader(Path(__file__).resolve().parent / 'prompts'),
    undefined=StrictUndefined,
    autoescape=False,
    keep_trailing_newline=True,
    auto_reload=False,
)
_TEMPLATES = frozenset(('audio-description.system.j2', 'audio-description.user.j2'))


def render_prompt(name, **context):
    if name not in _TEMPLATES:
        raise ValueError('등록되지 않은 프롬프트입니다')
    return _ENVIRONMENT.get_template(name).render(**context)

File: server/src/features/music-filter/prompt.renderer.js
const path = require('node:path');
const nunjucks = require('nunjucks');

// LLM용 일반 텍스트다. 입력값은 다시 템플릿으로 해석하지 않는다.
const environment = new nunjucks.Environment(
  new nunjucks.FileSystemLoader(path.join(__dirname, 'prompts'), { noCache: false }),
  { autoescape: false, throwOnUndefined: true },
);
const templates = new Set([
  'music-filter.system.njk', 'music-filter.user.njk',
  'public-guide.system.njk', 'public-guide.user.njk',
]);

function renderPrompt(name, context = {}) {
  if (!templates.has(name)) throw new Error('등록되지 않은 프롬프트입니다');
  return environment.render(name, context);
}

module.exports = { renderPrompt };
